### PR TITLE
refactor: split `Document` into `TextDocument` and `VectorDocument`

### PR DESCRIPTION
**Pull request by chr-hertel**
### Based on
* #95
* #101 

### Goal

`TextDocument` and `VectorDocument` - both have an UUID and Metadata

```php
/**
 * @param TextDocument|TextDocument[] $documents
 */
DocumentEmbedder::create(TextDocument|array $documents, ...): void

// only converting than

$vectorDocuments[] = new VectorDocument($vectors[$i], $document->id, $document->metadata);

// and drop text in store context completely

/**
 * @param VectorDocument|VectorDocument[] $documents
 **/
StoreInterface::add(VectorDocument ...$documents): void;

/**
 * @return VectorDocument[]
 */
VectorStoreInterface::query(Vector $vector, array $options = []): array;
```

### CODE REVIEW

File: src/DocumentEmbedder.php [hunk 1]
```diff
@@ -45,14 +43,14 @@ public function embed(Document|array $documents, int $chunkSize = 0, int $sleep
...
 
-            $vectorizedDocuments = [];
+            $vectorDocuments = [];
             foreach ($chunk as $i => $document) {
-                $vectorizedDocuments[] = $document->withVector($vectors[$i]);
+                $vectorDocuments[] = new VectorDocument($document->id, $vectors[$i], $document->metadata);
             }
 
-            $this->store->addDocuments($vectorizedDocuments);
+            $this->store->addDocuments($vectorDocuments);
```
Comment (OskarStark) [inline src/DocumentEmbedder.php:53]:
```suggestion
            $this->store->add($vectorDocuments);
```

File: src/Store/Azure/SearchStore.php [hunk 2]
```diff
@@ -26,21 +26,17 @@ public function __construct(
     ) {
     }
 
-    public function addDocument(Document $document): void
+    public function add(array|VectorDocument $documents): void
```
Comment (OskarStark) [inline src/Store/Azure/SearchStore.php:29]:
```suggestion
    public function add(VectorDocument ...$documents): void
```
?

File: src/Store/Azure/SearchStore.php [hunk 3]
```diff
@@ -73,23 +69,23 @@ private function request(string $endpoint, array $payload): array
     /**
      * @return array<string, mixed>
      */
-    private function convertDocumentToIndexableArray(Document $document): array
+    private function convertDocumentToIndexableArray(VectorDocument $document): array
```
Comment (OskarStark) [inline src/Store/Azure/SearchStore.php:72]:
```suggestion
    private function convertToIndexableArray(VectorDocument $document): array
```

File: src/Store/Azure/SearchStore.php [hunk 4]
```diff
@@ -73,23 +69,23 @@ private function request(string $endpoint, array $payload): array
...
             $this->vectorFieldName => $document->vector->getData(),
+            'text' => $document->text,
         ], $document->metadata->getArrayCopy());
     }
 
     /**
      * @param array<string, mixed> $data
      */
-    private function convertArrayToDocument(array $data): Document
+    private function convertArrayToDocument(array $data): VectorDocument
```
Comment (OskarStark) [inline src/Store/Azure/SearchStore.php:84]:
```suggestion
    private function convertToVectorDocument(array $data): VectorDocument
```

File: src/Store/Azure/SearchStore.php [hunk 5]
```diff
@@ -73,23 +69,23 @@ private function request(string $endpoint, array $payload): array
...
     {
-        return new Document(
+        return new VectorDocument(
             id: Uuid::fromString($data['id']),
-            text: null,
-            vector: null,
+            vector: $data[$this->vectorFieldName] ? new Vector($data[$this->vectorFieldName]) : null,
```
Comment (OskarStark) [inline src/Store/Azure/SearchStore.php:88]: I would assume, that the vector is available, I would assume the data was added via our Store classes

File: src/ToolBox/Tool/SimilaritySearch.php [hunk 6]
```diff
@@ -13,7 +13,7 @@
 final class SimilaritySearch
 {
     /**
-     * @var Document[]
+     * @var TextDocument[]
```
Comment (OskarStark) [inline src/ToolBox/Tool/SimilaritySearch.php:16]:
```suggestion
     * @var VectorDocument[]
```
?

File: src/Store/Azure/SearchStore.php (hunk 2 above)
Comment (chr-hertel) [reply src/Store/Azure/SearchStore.php:29]: yes, nice. one sec ...

File: src/DocumentEmbedder.php (hunk 1 above)
Comment (chr-hertel) [reply src/DocumentEmbedder.php]: yup, missed that

**Comment by OskarStark** [comment]
I like this PR

File: src/Store/Azure/SearchStore.php (hunk 5 above)
Comment (chr-hertel) [reply src/Store/Azure/SearchStore.php:87]: no, wait, it's not necessarily retrievable

**Comment by chr-hertel** [comment]
from my side ready to merge - any final thoughts?

**Review by OskarStark — COMMENTED** [review]
Some small things, afterwards good to squash merge!

File: src/Document/TextDocument.php [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+namespace PhpLlm\LlmChain\Document;
+
+use Symfony\Component\Uid\Uuid;
+use Webmozart\Assert\Assert;
+
+final readonly class TextDocument
+{
+    public function __construct(
+        public Uuid $id,
+        public string $text,
```
Comment (OskarStark) [inline src/Document/TextDocument.php:14]: One last thing! I think I would rename it:
```suggestion
        public string $content,
```

It is the "content" of the text document .... 🤔

**Review by OskarStark — APPROVED** [review]
Great job!

**Pull request merged**